src/prometheus-net.DotNetRuntime/DotNetRuntimeStatsBuilder.cs

```diff
@@ -59,17 +64,37 @@ public class Builder
             /// <returns></returns>
             public IDisposable StartCollecting()
             {
-                if (DotNetRuntimeStatsCollector.Instance != null)
+#if PROMV2
+                return StartCollecting(TCollectorRegistry.Instance, registryIsDefault: true);
+#elif PROMV3
+                return StartCollecting(Metrics.DefaultRegistry, registryIsDefault: true);
+#endif
+            }
+
+            /// <summary>
+            /// Finishes configuration and starts collecting .NET runtime metrics. Returns a <see cref="IDisposable"/> that
+            /// can be disposed of to stop metric collection. 
+            /// </summary>
+            /// <param name="registry">Registry where metrics will be collected</param>
+            /// <returns></returns>
+            public IDisposable StartCollecting(TCollectorRegistry registry)
+            {
+                return StartCollecting(registry, registryIsDefault: false);
+            }
+
+            private IDisposable StartCollecting(TCollectorRegistry registry, bool registryIsDefault)
+            {
+                if (registryIsDefault && DotNetRuntimeStatsCollector.Instance != null)
```

Comment:
I'm happy to allow collectors running against multiple registries but can you please ensure that this method is only called once per `TCollectorRegistry`? To support this I think `DotNetRuntimeStatsCollector.Instance` would have to become a `Dictionary<TCollectorRegistry, DotNetRuntimeStatsCollector>` (with all write being synchronised).

Reply:
Okay, that sounds good. I can work on that in a little bit.

Reply:
Instance changed to dictionary and usage moved under lock statement. The check is also moved under lock to guard against a race condition.